src/frontend/pages.ts

```diff
@@ -0,0 +1,78 @@
+import { Request, RequestHandler } from "express";
+import { contractsParamsGetter } from "./handlers/contracts";
+import { invoicesParamsGetter } from "./handlers/invoices";
+import { invoiceParamsGetter } from "./handlers/invoice";
+import path from "node:path";
+
+export type PageParamsGetter<TParams> = (
+  request: Request<TParams, unknown, unknown, unknown>
+) => object;
```

Comment:
Maybe unnecessary but it would be nice to have a generic for the return type too, rather than just `object`, which will let you do weird stuff like 

```
type Thing = {
  something: object
}

const thing: Thing = {
  something: Number
}
```

Reply:
I've added return types now, which seem to tidy things up pretty well.